src/lib.rs

```diff
@@ -380,6 +445,9 @@ pub enum CoreError {
     /// When thrown from complete_task, it means you should poll for a new task, receive a new
     /// task token, and complete that task.
     UnhandledCommandWhenCompleting,
+    /// Indicates that underlying function returned Ok, but result type was incorrect.
+    /// This is likely a result of a bug and should never happen.
+    UnexpectedResult,
```

Comment:
Yeah I think this goes away if we get rid of top-level task type and only accept appropriate completions, so probably worth fixing those in this PR

Reply:
Makes sense, as we've agreed, merging without that part, and we'll follow up with the refactor.